au4/src/effects/effects_base/internal/effectexecutionscenario.cpp

```diff
@@ -59,7 +57,10 @@ muse::Ret EffectExecutionScenario::doPerformEffect(AudacityProject& project, con
     EffectManager& em = EffectManager::Get();
     Effect* effect = nullptr;
 
-    NotifyingSelectedRegion& selectedRegion = ViewInfo::Get(project).selectedRegion;
+    auto selectionCtrl = selectionController();
```

Comment:
just in case, the `selectionController()` call is quite fast, i.e. it does not need to be optimized (not saved in a variable), i.e. using a variable can only make sense if you need to reduce the length of the code or make it more readable.
`selectionController()` implementation: 
```
    const std::shared_ptr<I>& get() const
    {
        if (!m_i) {
            const std::lock_guard<std::mutex> lock(StaticMutex::mutex);
            if (!m_i) {
                static std::string_view module = "";
                m_i = _ioc(iocContext())->template resolve<I>(module);
            }
        }
        return m_i;
    }

    const std::shared_ptr<I>& operator()() const
    {
        return get();
    }
```

i.e. the second and subsequent times, it is simply a return of a reference to a class member

Reply:
I figured.
In this case, it was indeed for readability.
You may see other places where I do this not for readability, but if I have to debug a lot, I then don't have to step in and out of that getter you've pasted.